File: TREE/Java/GFG - COUNT LEAVES IN A BINARY TREE.java
// GFG - COUNT LEAVES IN BINARY TREE
// QUESTION LINK - "https://www.geeksforgeeks.org/problems/count-leaves-in-binary-tree/1"

class Solution {
    private void countLeafUtil(Node root, int[] count) {
        if (root == null) return;
        if (root.left == null && root.right == null) {
            count[0]++;
        }
        countLeafUtil(root.left, count);
        countLeafUtil(root.right, count);
    }

    int countLeaves(Node root) {
        int[] count = new int[1];
        countLeafUtil(root, count);
        return count[0];
    }
}

File: TREE/C++/GFG - COUNT LEAVES IN A BINARY TREE.cpp
// GFG - COUNT LEAVES IN BINARY TREE
// QUESTION LINK - "https://www.geeksforgeeks.org/problems/count-leaves-in-binary-tree/1"


class Solution {
  public:
    void Leaf(int &res, Node* root) {
        if(root == NULL) return;
        if(root->left == NULL && root->right == NULL)
            res++;
        Leaf(res, root->left);
        Leaf(res, root->right);
    }

    int countLeaves(Node* root) {
        int res = 0;
        Leaf(res, root);
        return res;
    }
};

